bliss/encoder/metrics.py

```diff
@@ -64,7 +64,43 @@ def match_catalogs(self, true_cat, est_cat):
         return matching
 
 
-class DetectionPerformance(Metric):
+class SourceFilterMetric(Metric):
+    def __init__(self, source_type_filter: str | None):
```

Comment:
How about setting `source_type_filter` to `"total"` by default rather than `None`? Then you wouldn't need a separate variable named `source_type_name`